R/epi_df.R

```diff
@@ -189,7 +189,10 @@ new_epi_df <- function(x = tibble::tibble(geo_value = character(), time_value =
 
   # Reorder columns (geo_value, time_value, ...)
   if (nrow(x) > 0) {
-    x <- x %>% relocate(all_of(c("geo_value", other_keys, "time_value")), .before = 1)
+    all_names <- names(x)
+    ukey_names <- c("geo_value", other_keys, "time_value")
+    value_names <- all_names[!all_names %in% ukey_names]
+    x <- x[c(ukey_names, value_names)]
```

Comment:
Was this noticeably faster? Good to know, if so.

Reply:
This particular change seemed to help a lot, but that was likely because it was avoiding group reconstruction from us calling dplyr_reconstruct.  Another change avoids dplyr_reconstruct which avoids some of that same overhead.  (I re-ordered some commits to batch together all these inner loop tidyeval replacements, so history probably doesn't reflect the original order/benefit.)

Looking at this change atop the (current history of) previous commits:
- This individual change makes `arrange_col_canonical` take 20% of the old time.
- This individual change makes the epix_epi_slide_opt WIP take 97% of its old time.
- The tidyeval replacements together make the epix_epi_slide_opt WIP take 60% of the old time.